docs/logging.md

```diff
@@ -0,0 +1,165 @@
+It looks very similar to interfaces we all are used to (from SLF4J and other logging libraries and facades), introducing some new concepts:
+```scala mdoc
+import tofu.logging.LoggedValue
+
+trait LoggingBase[F[_]] {
+    def trace(message: String, values: LoggedValue*): F[Unit]
+    def debug(message: String, values: LoggedValue*): F[Unit]
+    def info(message: String, values: LoggedValue*): F[Unit]
+    def warn(message: String, values: LoggedValue*): F[Unit]
+    def error(message: String, values: LoggedValue*): F[Unit]
+}
+```
+
+Basically, that's the main user API you can interact with throughout your code. 
+It resembles well-known solutions, keeping in mind that logging is an effect, though introducing the concept of `LoggedValue`.  
+
+
+
+#### Logging creation
+Creating an instance of `Logging` in Tofu is an effect itself. You can use helper class `Logs` that can produce values of
+type `I[Logging[F]]` (where `I` and `F` can be two different effects) and define a behaviour of your `Logging` instances, 
+for example:  
+```scala mdoc
+import tofu.logging._
+
+// for simplification, you can use whatever F you like
+type F[A] = cats.effect.IO[A]
+
+class MyService
+
+// defines simple logging behaviour without any notion of contexts
+val syncLogs = Logs.sync[F, F]
+
+// describing creation of logging instances and logging itself. When being run, this will log a message
+val effect: F[Unit] = for {
+    loggingBN <- syncLogs.byName("my-logger") // you can create instances by name
+    loggingFS <- syncLogs.forService[MyService] // you can create instances by type tags
+    _         <- loggingBN.info("This is an INFO message")
+    _         <- loggingFS.warn("This is a WARN message")
+} yield ()
+```
+
+There are multiple ways to create `Logs` instance, among them:
+* `Logs.sync`, requiring a `cats.effect.Sync` instance for delaying side-effects
+* `Logs.withContext`, requiring a `cats.effect.Sync` instance for delaying side-effects and a proof that `F` has a context to log
+* `Logs.const` that wraps single `Logging` thus always returning the same instance
+* `Logs.empty` that will produce a no-op `Logging` instances
+* `Logs.combine` that can combine two instances of `Logs` thus combining produces instances
+
+
+#### Context logging
+It is possible to log the context of your computations (MDC, remember?) if you have an instance of `tofu.Context` for your
+effect type and a `Loggable` for your context. 
+That's natural - describe how we can obtain a context from a computation `F` and a way to represent that context in log, 
+and we will ensure it will be logged anytime you use `Logging` operations. 
+You can log that context to string as well as use it as a part of structured logging. 
+
+#### What LoggedValue and Loggable is
+`LoggedValue` is a representation of a value that can be written to a log in a string form. 
+It's used internally, and you do not need to use it directly most of the time. Instead, what you want is a `Loggable` 
```

Comment:
> It's used internally, and you do not need to use it directly most of the time.

So why do we start talking about `LoggedValue`?

Reply:
Because user API showcases functions like `def trace(message: String, values: LoggedValue*)` and it may be unclear why do we define `Loggable` instances when API requires `LoggedValue`s

Reply:
> Because user API showcases functions 

API is separated from the sentence by subtitle. Context is lost.

I suggest teaching a user how to
* log objects
* format logs
and not what some classes/traits can do. Think about it like inversion of accents.